plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/status-filter/block.json

```diff
@@ -74,6 +74,26 @@
 		"clearButton": {
 			"type": "boolean",
 			"default": true
+		},
+		"onsale": {
+			"type": "boolean",
+			"default": false
+		},
+		"featured": {
+			"type": "boolean",
+			"default": false
+		},
+		"instock": {
+			"type": "boolean",
+			"default": false
+		},
+		"outofstock": {
+			"type": "boolean",
+			"default": false
+		},
+		"onbackorder": {
+			"type": "boolean",
+			"default": false
```

Comment:
I've pushed up changes to dynamically pull in the stock statuses. With this change, we will have to make some adjustments.

- Because the settings are now dynamic, by default, we should not enable any of the statuses. Let the user choose on their own.
- We need to remove the real counts from the editor preview because we may not know this information ahead of time. We can just use placeholder data.
- Currently blocks are registered via the block.json. Because we're injecting dynamic attributes, we can no longer use the block.json for the attributes. But the side effect with this is that in the server side render, we can't get the $attributes data anymore. We need to come up with a solution for this.

Reply:
One idea is to not use `block.json` just for this block and overload the `register_block_type` method. I have explore if that would work.

Reply:
> One idea is to not use block.json just for this block and overload the register_block_type method. I have to explore if that would work.

Yeah, we're doing this in some blocks, like this one https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/active-filters/index.tsx

> Because the settings are now dynamic, by default, we should not enable any of the statuses. Let the user choose on their own.

I have a different opinion. I think we should enable all by default and let them check off what they don't need. By doing so, the block "works" right after it's added to the editor. Having everything checked off can give people a sense that the block is not working, IMO.

Reply:
> Yeah, we're doing this in some blocks, like this one

I don't think that works. On server side, you won't have access to the attributes that are not declared in the `block.json`.

> I have a different opinion. I think we should enable all by default and let them check off what they don't need. By doing so, the block "works" right after it's added to the editor. Having everything checked off can give people a sense that the block is not working, IMO.

This is fine, I don't have strong opinion either way.

Reply:
Pushed up updates to use the `register_block_type` method I mentioned. This works. I also defaulted all statuses to be enabled.